acsoo/addons_toupdate.py

```diff
@@ -0,0 +1,152 @@
+            ['git', 'show', diff_req_ref], stderr=devnull)
+    except click.ClickException:  # The requirements file is new
+        return 'all'
+    # If requirements are the same, stop
+    if requirements_string == diff_requirements_string:
+        return dependencies
+    # Parse the requirements files
+    current_requirements = {
+        req.name.replace('-', '_'):
+            req for req in requirements.parse(requirements_string)}
+    diff_requirements = {
+        req.name.replace('-', '_'):
+            req for req in requirements.parse(diff_requirements_string)}
+    # Compare the two requirements files and populate modified addons list
+    for req_name in current_requirements:
+        current_req = current_requirements.get(req_name)
+        diff_req = diff_requirements.get(req_name)
+        if not diff_req:  # New dependency, ignore
+            continue
+        # Special case for odoo and enterprise addons, update all if change
+        if ODOO_REGEX.match(req_name) and \
+                (current_req.specs != diff_req.specs or
+                 current_req.revision != diff_req.revision):
+            return 'all'
+        # Special case for external sources, update all if change
+        # TODO: recursive comparison of the changes in the external sources
+        if ODOO_ADDONS_REGEX.match(req_name) and \
+                (current_req.specs != diff_req.specs or
+                 current_req.revision != diff_req.revision):
+            return 'all'
+        if _odoo_addon_dependency_changed(req_name, current_req, diff_req):
+            dependencies.add(
+                ODOO_ADDON_REGEX.match(req_name).group('addon_name'))
+    return dependencies
+
+
+@click.command(help="Print a comma separated list of the modified installable "
+                    "addons from the given git ref")
+@click.argument('git_ref')
+@click.option('-r', '--diff-requirements', 'diff_requirements', is_flag=True,
+              help="Defines whether the comparison must take the requirements "
+                   "file into account or not.")
+@click.option('--exclude', default='',
+              help="Comma separated list of addons to exclude from "
+                   "the dependencies.")
+@click.option('--requirement', default='requirements.txt',
+              type=click.Path(dir_okay=False, exists=True),
+              help='Requirements to use for requirements comparison '
+                   '(default=requirements.txt)')
+@click.pass_context
+def addons_toupdate(ctx, git_ref, diff_requirements, exclude, requirement):
+    # Check ancestor
+    if call(['git', 'merge-base', '--is-ancestor', git_ref, 'HEAD']):
+        click.echo('all')
+        return
+    addon_names = set()
+    # Compare each installable addons and populate modified addons set
+    addons = ctx.obj['addons']
+    for addon_name, (addon_dir, manifest) in addons.items():
+        if call(['git', 'diff', '--quiet', '--diff-filter=M',
```

Comment:
I think this diff-filter thing is not correct. What if for instance you add a file in an existing addon?